### PR TITLE
Code fixes from PR 34997

### PR DESCRIPTION
**Pull request by come-nc**
Split code fixes commits from https://github.com/nextcloud/server/pull/34997 to make the diff shorter.

### CODE REVIEW

File: apps/dav/lib/Connector/Sabre/File.php [hunk 1]
```diff
@@ -273,9 +273,9 @@ public function put($data) {
 			if ($result === false) {
 				$expected = -1;
 				if (isset($_SERVER['CONTENT_LENGTH'])) {
-					$expected = $_SERVER['CONTENT_LENGTH'];
+					$expected = (int)$_SERVER['CONTENT_LENGTH'];
```
Comment (nickvergessen) [inline apps/dav/lib/Connector/Sabre/File.php:276]: Will we run into issues with 32bit and float?

**Review by nickvergessen — CHANGES_REQUESTED** [review]
Good apart from the (int) casting on file size (content length) which makes my alarms ring

**Review by YasinAshorian — APPROVED** [review]

File: apps/dav/lib/Connector/Sabre/File.php (hunk 1 above)
Comment (come-nc) [reply apps/dav/lib/Connector/Sabre/File.php:276]: No I don’t think so, I never saw an HTTP request with a size larger than 4GB
Content length should always fit an int I think.
And there is already the same cast on line 295.

Comment (nickvergessen) [reply apps/dav/lib/Connector/Sabre/File.php:276]: Ah, nevermind then

**Review by nickvergessen — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**